crates/red_knot/src/symbols.rs

```diff
@@ -33,6 +32,32 @@ pub fn symbol_table(db: &dyn SemanticDb, file_id: FileId) -> QueryResult<Arc<Sym
     })
 }
 
+#[tracing::instrument(level = "debug", skip(db))]
+pub fn resolve_global_symbol(
+    db: &dyn SemanticDb,
+    module: ModuleName,
+    name: &str,
+) -> QueryResult<Option<GlobalSymbolId>> {
+    let Some(typing_module) = resolve_module(db, module)? else {
+        return Ok(None);
+    };
+    let typing_file = typing_module.path(db)?.file();
+    let typing_table = symbol_table(db, typing_file)?;
+    let Some(typing_override) = typing_table.root_symbol_id_by_name(name) else {
+        return Ok(None);
+    };
+    Ok(Some(GlobalSymbolId {
+        file_id: typing_file,
+        symbol_id: typing_override,
+    }))
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
```

Comment:
I think it's okay to add  `Eq` as well.

```suggestion
#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```